src/main/java/Duke.java

```diff
@@ -1,10 +1,195 @@
+    private ToDo newToDo;
+    private Event newEvent;
+    private Deadline newDeadline;
+
+    public Duke() {
+        this.tasks = new ArrayList<Task>();
+    }
+
+    private void printLine() {
+        String boundary = " ======================================================================";
+        System.out.println(boundary);
+    }
+
+    private void printGreetingMessage() {
+        String firstGreetingMessage = " Hi! I am Duke, your next doooooorrrr friendly elf.....I mean bot";
+        String secondGreetingMessage = " How may Dukeeeeee help you today?";
+        String line = "_______________________________________________________________________";
+        System.out.println(firstGreetingMessage);
+        System.out.println(secondGreetingMessage);
+        System.out.println(line);
+    }
+
+    private void printLeavingMessage() {
+        String leavingMessage = " Bye! Duke is now a freeeeee elf again!!!!";
+        System.out.println(leavingMessage);
+    }
+
+    private void printList() {
+        int position = 1;
+        for (Task text: tasks) {
+            System.out.println("    " + position + ". " + text);
+            position ++;
+        }
+    }
+
+    private void list() {
+        if (tasks.size() == 0) {
+            System.out.println(" [Warning: There are currently no tasks!]");
+        } else {
+            System.out.println(" Here are the list of tasks:");
+            printList();
+        }
+    }
+
+    private int getIndex(String input) {
+        String removeTrailingSpaces = input.trim();
+        String numericalIndex = removeTrailingSpaces.substring(5,input.length()).trim();
+        int index = Integer.valueOf(numericalIndex) - 1;
+        return index;
+    }
+
+    private void markAsDone(int index) {
+        Task completedTask = tasks.get(index);
+        boolean isCompleted = completedTask.getStatus();
+        if (isCompleted) {
+            System.out.println(" [Warning: The task(s) has already been marked as done]");
+        } else {
+            completedTask.markAsDone();
+            tasks.set(index, completedTask);
+            System.out.println(" Duke has marked the following tasks as done:");
+            System.out.println("  " + completedTask);
+        }
+    }
+
+    private void getDoneExceptions(String input) {
+        try {
+            int index = getIndex(input);
+            markAsDone(index);
+        } catch(IndexOutOfBoundsException exception) {
+            System.out.println(" [Error: " + exception + "]");
+            System.out.println(" Please check your input!");
+        }
+    }
+
+    private void addToDo(String input) {
+        String removeTrailingSpaces = input.trim();
+        String description = removeTrailingSpaces.substring(5, input.length());
+        newToDo = new ToDo(description);
+        tasks.add(newToDo);
+        System.out.println(" Got it! I have added the following tasks: ");
+        System.out.println( newToDo);
```

Comment:
perhaps you can change System.out.println( newToDo) to System.out.println(newToDo) for code consistency.